erpnext/selling/doctype/customer/customer.js

```diff
@@ -103,6 +103,31 @@ frappe.ui.form.on("Customer", {
 		}
 	},
 
+	onload: function (frm) {
+		let days_of_week = moment.weekdays()
+		let fields = []
+		days_of_week.forEach(day => {
+			fields.push(
+				{
+					"label": __(day),
+					"value": day,
+					"checked": frm.doc.delivery_days ? frm.doc.delivery_days.split(",").includes(day) : 0
```

Comment:
If you have an array, this is easier to compare:

```suggestion
					"checked": frm.doc.delivery_days ? frm.doc.delivery_days.includes(day) : 0
```